### PR TITLE
Simplify the installation of man pages

### PR DESCRIPTION
**Pull request by shindere**
This PR changes the extension of manual pages stored in the repository
from `.m` to `.1` and adds the three ones that were produced with `echo`
during `make install` as real files.

It also gets rid of the variables containing the names of manual sections
for programs and libraries because that does not look really useful.

Finally, the directory where manual pages for commands are installed is
now created by `man/Makefile` rather than the root `Makefile` and
the errors that could happen while running `make -C man install` are
no longer ignored by the root makefile.

@dra27 and/or @Octachron may want to review this, which is a quite
straightforward PR, I think.

### CODE REVIEW

**Review by Octachron — APPROVED** [review]
The hard coding of the man section numbers seem reasonable, since I am not aware of OS using man pages with different section numbers.

Overall, this looks like a worthwhile simplification of the build system.

File: Changes [hunk 1]
```diff
@@ -257,6 +257,9 @@ Working version
 
 ### Build system:
 
+- #10717 Simplify the installation of man mages
```
Comment (dra27) [inline Changes:260]: Nits only: needs a colon after #10717 and pages rather than mages

**Review by dra27 — COMMENTED** [review]
This is much nicer, yes. One thought on the `.so` link files, which I think I'd spotted in passing before - we do links for `ocamlc.opt`, `ocamlopt.opt` and `ocamloptp`, but these days we have many other `.opt` versions (and also the `.byte` versions).

Is that worth cleaning up at some point and, with more than 3 of them, would we really want to generate those (clean-ups include the fact that we install ocamldoc in bytecode and ocamldoc.opt, for example).

**Comment by shindere** [comment]
I went ahead and merged. Thanks.

**Comment by shindere** [comment]
David Allsopp (2021/10/21 15:29 +0000):
> @dra27 commented on this pull request.
>
> This is much nicer, yes. One thought on the `.so` link files, which I think I'd spotted in passing before - we do links for `ocamlc.opt`, `ocamlopt.opt` and `ocamloptp`, but these days we have many other `.opt` versions (and also the `.byte` versions).
>
> Is that worth cleaning up at some point and, with more than 3 of them,
> would we really want to generate those (clean-ups include the fact
> that we install ocamldoc in bytecode and ocamldoc.opt, for example).

Perhaps we shoudl be way more disciplined in how and what we isntall,
yes. For instance at the moment we install manual pages independently of
what has been enabled / configured. I'm not sure either we have a man
page for eachi binary file we install, perhaps this could be checked by
CI somehow?

Is all that worth an issue to keep track of it?

> > @@ -257,6 +257,9 @@ Working version
>
>  ### Build system:
>
> +- #10717 Simplify the installation of man mages
>
> Nits only: needs a colon after #10717 and pages rather than mages

Thanks! Fixed in 5915a72545.

**Comment by Et7f3** [comment]
> review by ???

**Comment by shindere** [comment]
Changes entry fixed in d0129e2914a8862f5ecd0310200ff22bc089f38b

**Pull request merged**